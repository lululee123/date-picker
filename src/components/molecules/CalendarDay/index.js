import React from 'react';
import dayjs from 'dayjs';

import Day from 'components/atoms/Day';

import styles from './index.scss';

const CalendarDay = ({ days = [], selected = '', today = '', currentMonth = '', onSelect = () => {} }) => {
	const detectType = day => {
		if (day === selected) {
			return 'active';
		}

		if (day === today) {
			return 'today';
		}

		if (dayjs(day).format('MM') !== currentMonth) {
			return 'inActive';
		}

		return 'normal';
	};

	const formatDate = day => (dayjs(day).format('YYYY-MM-DD'));

	return (
		<div className={styles.calendarDay}>
			{days.map(row => (
				row.map(day => (
					<Day
						key={day}
						value={formatDate(day)}
						displayText={dayjs(day).format('D')}
						type={detectType(formatDate(day))}
						onClick={onSelect}
					/>
				))
			))}
		</div>
	);
};

export default CalendarDay;
