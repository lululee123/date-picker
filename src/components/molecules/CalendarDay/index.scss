.calendarDay {
    display: flex;
    flex-wrap: wrap;
}