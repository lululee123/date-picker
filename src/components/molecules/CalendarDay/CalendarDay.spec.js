import React from 'react';
import renderer from 'react-test-renderer';

import CalendarDay from './index';

test('render CalendarDay correctly', () => {
	const component = renderer
		.create(
			<CalendarDay />,
		)
		.toJSON();
	expect(component).toMatchSnapshot();
});
