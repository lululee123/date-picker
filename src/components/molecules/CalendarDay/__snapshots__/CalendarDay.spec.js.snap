// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render CalendarDay correctly 1`] = `
<div
  className="calendarDay"
/>
`;
